packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1715,9 +1715,9 @@ class _DropdownButtonFormFieldState<T> extends FormFieldState<T> {
 
   @override
   void didChange(T? value) {
-    super.didChange(value);
     assert(widget.onChanged != null);
     widget.onChanged!(value);
+    super.didChange(value);
```

Comment:
It is the same  for _CupertinoTextFormFieldRowState and _TextFormFieldState

Reply:
Nice caught.